ui/ducks/confirm-transaction/confirm-transaction.duck.js

```diff
@@ -308,3 +344,38 @@ export function clearConfirmTransaction() {
     type: CLEAR_CONFIRM_TRANSACTION,
   };
 }
+
+export const updateTxValueIfMaxEthSettled = (
+  dispatch,
+  state,
+  { gasFeeEstimates, gasEstimateType },
+) => {
+  const txId = state.confirmTransaction?.txData?.id;
+
+  if (
+    txId &&
+    state.confirmTransaction.sendTxDetailPerId?.[txId]?.amountMode ===
+      AMOUNT_MODES.MAX &&
+    state.confirmTransaction.sendTxDetailPerId?.[txId]?.assetType ===
+      AssetType.native
+  ) {
+    const selectedAccount = getSelectedAccount(state);
+    const gasEstimationObject = getGasEstimationObject(
+      state,
+      { gasFeeEstimates, gasEstimateType },
+      state.confirmTransaction.txData,
```

Comment:
This chain is repeated to compute txId and also below to compute transactionMeta. It might be more readable to store it in an intermediate variable at the top of the function.

`const txData = state.confirmTransaction.txData``